lib/chef/action_collection.rb

```diff
@@ -223,6 +226,7 @@ def resource_failed(new_resource, action, exception)
 
       current_record.status = :failed
       current_record.exception = exception
+      current_record.error_description = Formatters::ErrorMapper.resource_failed(new_resource, action, exception).for_json
```

Comment:
Any reason we don't just use this:

https://github.com/chef/chef/blob/6e73617243baa21815258f9b51352e9f3753af1d/lib/chef/data_collector/error_handlers.rb#L110

Reply:
We don't have data collector object reference, also `error_description` gets assigned with the call of `resource_failed`.

please let me know if I am not getting it correctly? Thanks

Reply:
Yeah actually that's correct, the data_collector isn't responsible for per-resource analysis and while it gets the error_description after resource_failed() is called, it doesn't do anything with it, and as it collects more ignored errors during the run it just discards them.  To collect the errors in the data_collector for the actions we'd have to reimplement chunks of the action_collection there (violating SRP and defeating the purpose of extracting the action_collection).  So it does make more sense to collection the error descriptions in the action_record like this.